server/src/main/java/io/crate/exceptions/RoleAlreadyExistsException.java

```diff
@@ -23,9 +23,17 @@
 
 import java.util.Locale;
 
+import org.jetbrains.annotations.NotNull;
+
+import io.crate.role.JwtProperties;
+
 public class RoleAlreadyExistsException extends RuntimeException implements ConflictException, UnscopedException {
 
     public RoleAlreadyExistsException(String roleName) {
         super(String.format(Locale.ENGLISH, "Role '%s' already exists", roleName));
     }
+
+    public RoleAlreadyExistsException(String roleName, @NotNull JwtProperties jwtProperties) {
+        super(String.format(Locale.ENGLISH, "Role '%s' or another role with the same combination of jwt properties already exists", roleName));
+    }
```

Comment:
Having `jwtProperties` unused to use a different ctor looks not great.
Maybe use a factory method and add a default private ctor?
```suggestion
    public RoleAlreadyExistsException for(String roleName, @Nullable JwtProperties jwtProperties) {
        if (jwtProperties == null) {
            return new RoleAlreadyExistsException(String.format(Locale.ENGLISH, "Role '%s' already exists", roleName));
        } else {
            return new RoleAlreadyExistsException(String.format(Locale.ENGLISH, "Role '%s' or another role with the same combination of jwt properties already exists", roleName));
        }
    }
```

this way no `jwtProperties` null check must be done outside.

Reply:
Thanks, applied and adapted depending code